### PR TITLE
Feature: Fetching Identify parameters before harvesting ( if HARVESTI…

### PR DESCRIPTION
**Pull request by lareferencia**
…NG_IDENTIFY_PARAMETERS == TRUE )

This one solves issue  #3

Please note that lrharvester app must be updated in order to have the HARVESTING_IDENTIFY_PARAMETERS option in the UI

image

@paulo-graca Please test

### CODE REVIEW

**Comment by paulo-graca** [comment]
Thank you @lmatas I will try it out and let you know!

File: src/main/java/org/lareferencia/core/harvester/workers/HarvestingWorker.java [hunk 1]
```diff
@@ -137,7 +140,34 @@ public void stop() {
...
+
+		// date granularity
+		String granularity = DEFAULT_GRANDULARITY;
+
+		// check if the url is valid by trying to connect to it with identify verb
+		String originURL = runningContext.getNetwork().getOriginURL();
+
+
+		// if fetch identify is true, will try to fetch the identify information
+		if ( runningContext.getNetwork().getBooleanPropertyValue("HARVEST_IDENTIFY_PARAMETERS") ) {
```
Comment (paulo-graca) [inline src/main/java/org/lareferencia/core/harvester/workers/HarvestingWorker.java:155]: Does this makes sense? You are given the admin user the responsability to understand how the OAI of the repository works and add this setting on a different screen, you are creating an usability issue, the person when defining the OAI, must also remember to go to a different screen to check the required configuration. I don't think it's very praticle, when, like us, you have more than 400 resources.

**Review by paulo-graca — COMMENTED** [review]
This feature worked for us, but, as is, present us an usability challenge.
I think we should look into `HarvestingWorker` in a different perspective. Currently we need to set all the settings we need to activate for the harvesting. We are looking to this feature as an developer when we want to only address specific stuff, to have the must optimized process as possible.
But, for a new user, technical or not, this is difficult to achieve. I think we should address them differently. As a always active settings, letting the featured and skilled user to fine tune and disable what he don't need for that particular network, disabling it.

Saying this, I think we can have a new setting to control the Identify access, but be always active by default. I also think we should do this to all other settings in this Worker, leaving us to  only "activate" what we don't want. Meaning all settings should be renamed to a inverse, contrary or negative naming, also changing the logic in the worker.

**Review by paulo-graca — CHANGES_REQUESTED** [review]
I did another test and an error still occurs with incremental harvesting:


```
2022-10-19 10:32:09.848  INFO 50732 --- [nio-8090-exec-2] o.l.b.controllers.ActionsController      : HARVESTING_ACTION:: incremental:true :: network: scielopt(id:452)
2022-10-19 10:32:10.656  INFO 50732 --- [askScheduler-10] o.l.c.h.workers.HarvestingWorker         : Identify Granularity found: YYYY-MM-DD
2022-10-19 10:32:10.669  INFO 50732 --- [askScheduler-10] o.l.c.h.workers.HarvestingWorker         : scielopt(id:452) Starting incremental harvesting from date: 2022-10-19T09:59:30Z -- Snapshot: 23542
2022-10-19 10:32:10.675  INFO 50732 --- [askScheduler-10] o.l.c.h.workers.HarvestingWorker         : Harvesting is starting for scielopt(id:452)...
2022-10-19 10:32:10.690  INFO 50732 --- [askScheduler-10] o.l.c.h.workers.HarvestingWorker         : There aren't defined sets. Harvesting the entire collection for scielopt(id:452)
2022-10-19 10:32:10.711 ERROR 50732 --- [askScheduler-10] o.l.c.h.workers.HarvestingWorker         : Fatal Error:ListRecords call error :: error code: badArgument
2022-10-19 10:32:10.730 ERROR 50732 --- [askScheduler-10] o.l.c.h.workers.HarvestingWorker         : scielopt(id:452) :: Harvesting ended with errors !!!

```

**Comment by lmatas** [comment]
Question:
Do you know if Scielo supports incremental harvesting??
Lautaro Matas

**Comment by paulo-graca** [comment]
Yes. It supports. The request was made still ignoring the granularity:

```
2022-10-19 10:42:13.562 DEBUG 58080 --- [taskScheduler-3] o.o.oai.harvester2.verb.HarvesterVerb    : requestURL=https://www.scielo.pt/oai/scielo-oai.php?verb=ListRecords&from=2022-10-19T09:58:26Z&metadataPrefix=oai_dc
2022-10-19 10:42:13.562 DEBUG 58080 --- [taskScheduler-3] o.o.oai.harvester2.verb.HarvesterVerb    : try:0 getting requestURL: https://www.scielo.pt/oai/scielo-oai.php?verb=ListRecords&from=2022-10-19T09:58:26Z&metadataPrefix=oai_dc
2022-10-19 10:42:13.598 DEBUG 58080 --- [taskScheduler-3] o.o.oai.harvester2.verb.HarvesterVerb    : https://www.scielo.pt/oai/scielo-oai.php?verb=ListRecords&from=2022-10-19T09:58:26Z&metadataPrefix=oai_dc responseCode=200
2022-10-19 10:42:13.598 DEBUG 58080 --- [taskScheduler-3] o.o.oai.harvester2.verb.HarvesterVerb    : contentEncoding=null

```

This doesn't work:
https://www.scielo.pt/oai/scielo-oai.php?verb=ListRecords&from=2022-10-19T09:58:26Z&metadataPrefix=oai_dc

but this does:
https://www.scielo.pt/oai/scielo-oai.php?verb=ListRecords&from=2022-10-19&metadataPrefix=oai_dc

**Comment by lmatas** [comment]
Ok, I will test with this exact case and let you know.
Lautaro Matas

**Comment by lareferencia** [comment]
***@***.*** The problem was a missing granularity case, I didn't
consider the case sensitive issue on the first implementation. Now I think
that will work, please test. *

Lautaro Matas

**Review by paulo-graca — APPROVED** [review]
Now it works as expected.

**Pull request merged**